### PR TITLE
Bump msw

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Build on top of https://github.com/osbuild/image-builder-frontend/pull/2186 and in the last commit:

* Add origin to have absolute path in MSW
* Fix the spy handler event to make sure it works as expected
* Also bumps few dependencies, but IMO that's not needed.

### CODE REVIEW

File: src/store/emptyImageBuilderApi.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { IMAGE_BUILDER_API } from '../constants';
 export const emptyImageBuilderApi = createApi({
   reducerPath: 'imageBuilderApi',
   baseQuery: fetchBaseQuery({
-    baseUrl: IMAGE_BUILDER_API,
+    baseUrl: window.location.origin + IMAGE_BUILDER_API,
```
Comment (ezr-ondrej) [inline src/store/emptyImageBuilderApi.ts:9]: this was the culprit of the errors.

File: src/test/Components/CreateImageWizard/wizardTestUtils.tsx [hunk 2]
```diff
@@ -17,8 +16,9 @@ type RequestTypes = 'GET' | 'PUT' | 'POST' | 'DELETE';
 
 export function spyOnRequest(pathname: string, method: RequestTypes) {
   return new Promise((resolve) => {
-    const listener = async (req: MockedRequest) => {
-      if (req.url.pathname === pathname && req.method === method) {
+    const listener = async ({ request: req, requestId }: { request: Request, requestId: string }) => {
```
Comment (ezr-ondrej) [inline src/test/Components/CreateImageWizard/wizardTestUtils.tsx:19]: this listener also needed to be fixed, as the structure was changed.

File: src/test/setup.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import nodeFetch, { Request, Response } from 'node-fetch';
+// import nodeFetch, { Request, Response } from 'node-fetch';
 
 import { server } from './mocks/server';
 import 'vitest-canvas-mock';
 
-Object.assign(global, { fetch: nodeFetch, Request, Response });
+// Object.assign(global, { fetch: nodeFetch, Request, Response });
```
Comment (ezr-ondrej) [inline src/test/setup.ts:6]: I've removed this and it changed nothing, no idea why it's here, any idea @regexowl ?
But IMO it is irrelevat to the fix.

File: src/test/fixtures/composes.ts [hunk 4]
```diff
@@ -26,10 +24,8 @@ export const mockComposesEmpty: ComposesResponse = {
 const currentDate = new Date();
 const currentDateInString = currentDate.toISOString();
 
-export const composesEndpoint = (
-  req: RestRequest<never, PathParams<string>>
-) => {
-  const params = req.url.searchParams;
+export const composesEndpoint = (url: URL) => {
+  const params = new URLSearchParams(url.search);
```
Comment (ezr-ondrej) [inline src/test/fixtures/composes.ts:28]: This fixes the ImagesTable tests - by fixing the composes endpoint.

File: src/test/setup.ts (hunk 3 above)
Comment (regexowl) [reply src/test/setup.ts]: I've added the polyfills during the vitest migration as there were errors outputted without them. But it doesn't seem like the case anymore with bumped msw, so I'd say we can get rid of them :+1:

**Comment by ezr-ondrej** [comment]
https://github.com/osbuild/image-builder-frontend/pull/2186 is in :tada:

**Pull request closed without merging**